fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -927,10 +947,32 @@ ACTOR Future<Void> getResolution(CommitBatchContext* self) {
 			TenantInfo const& tenantInfo = trs[t].tenantInfo;
 			int64_t tenantId = tenantInfo.tenantId;
 			Optional<TenantNameRef> const& tenantName = tenantInfo.name;
-			// TODO(yiwu): In raw access mode, use tenant prefix to figure out tenant id for user data
 			if (tenantId != TenantInfo::INVALID_TENANT) {
 				ASSERT(tenantName.present());
 				encryptDomains[tenantId] = tenantName.get();
+			} else if (!trs[t].transaction.mutations.empty()) {
+				// For raw-access mode, parse 'tenantId' from transaction mutations by accessing first 8 bytes.
+				// For a given transaction, participant mutations can't cross tenant boundaries.
```

Comment:
Is this property enforced somewhere?

Reply:
I did see clear ranges that can cross tenant boundary in simulation before.